exposed-core/src/main/kotlin/org/jetbrains/exposed/dao/id/IdTable.kt

```diff
@@ -34,6 +34,71 @@ object EntityIDFunctionProvider {
 abstract class IdTable<T : Comparable<T>>(name: String = "") : Table(name) {
     /** The identity column of this [IdTable], for storing values of type [T] wrapped as [EntityID] instances. */
     abstract val id: Column<EntityID<T>>
+
+    /** All base columns that make up this [IdTable]'s identifier column. */
+    val idColumns = HashSet<Column<out Comparable<*>>>()
```

Comment:
Probably we should validate that user does not add more than one column for non composite tables, otherwise he can create such a table and it would work:

```
@Test
fun test() {
    val tester = object : IntIdTable("test") {
        val field = integer("field").entityId()
        val field2 = integer("field2").entityId()
    }

    withTables(tester) {
        tester.insert {
            it[field] = 1
            it[field2] = 2
        }
    }
}
```

Reply:
This is one of my concerns about `entityId()`. Currently there is nothing stopping a user from creating the table in your example and working with it anyway. So relying on it to be the source of truth for which columns are component key columns might be dependent on whether users have been creating entity id columns like this way.

How would you suggest to validate this best?

Reply:
It's possible that the user wants to create a table with a composite key, but does not know that a special table class exists for it. From another side it's possible that the user needed a table with one primary key, but added two by mistake. I think that throwing exception with a recommendation either to change table class to composite one, or to leave only one id would be sufficient. It will be thrown on the starting the application, so it should be fine.

I don't expect that users create table objects dynamically >_<